docs/API.md

```diff
@@ -1726,67 +609,70 @@ loss list is empty and the flight window is full, all packets since the last
-the same order in which they were sent. In the situation where a 
-message suffers a packet loss, this prevents any subsequent messages
-from achieving completion status prior to recovery of the preceding message.
-
-The sending function will HANGUP when the free space in the sending
-buffer does not exactly fit the whole message, and it will only RESUME
-if the free space in the sending buffer grows up to this size. The
-call to the sending function also returns with an error, when the
-size of the message exceeds the total size of the buffer (this can
-be modified by `SRTO_SNDBUF` option). In other words, it is not
-designed to send just a part of the message -- either the whole message
-is sent, or nothing at all.
-
-The receiving function will HANGUP until the whole message is available
-for reading; if the message spans multiple UDP packets, then the function
-RESUMES only when every single packet from the message has been
-received, including recovered packets, if any. When the INORDER flag is set
-to false and parts of multiple messages are currently available,
-the first message that is complete (possibly recovered) is returned. 
-Otherwise the function does a HANGUP until the next message
-is complete. The call to the receiving function is rejected if the
-buffer size is too small for a single message to fit in it.
-
-Note that you can use any of the sending and receiving functions
-for sending and receiving messages, except sendfile/recvfile, which
-are dedicated exclusively for Buffer API. 
+**HISTORICAL INFO**: The library on which SRT was based (UDT) somewhat misleadingly
+used the terms `STREAM` and `DGRAM`, and used the system symbols `SOCK_STREAM` and 
+`SOCK_DGRAM` in the socket creation function. A "datagram" in the UDT terminology 
+has nothing to do with the "datagram" term in networking terminology, where its 
+size is limited to as much it can fit in one MTU. In UDT it is actually a message, 
+which may span multiple UDP packets and has clearly defined boundaries. It's rather 
+similar to the **SCTP** protocol. Also, in UDP the API functions were strictly bound 
+to `DGRAM` or `STREAM` mode: `UDT::send/UDT::recv` were only for `STREAM` and
+`UDT::sendmsg/UDT::recvmsg` only for `DGRAM`. In SRT this is changed: all functions 
+can be used in all modes, except `srt_sendfile/srt_recvfile`, and how the functions 
+actually work is controlled by the `SRTO_MESSAGEAPI` flag.
+
+In message mode, every sending function sends **exactly** as much data as it is 
+passed in a single sending function call. The receiver also receives not less than 
+**exactly** the number of bytes that was sent (although every message may have a 
+different size). Every message may also have extra parameters:
+
+  - **TTL** defines how much time (in ms) the message should wait in the sending 
+  buffer for the opportunity to be picked up by the sender thread and sent over 
+  the network; otherwise it is dropped.
```

Comment:
There's one thing nonprecise here: The packet is ALWAYS sent for the first time. The TTL parameter sets only a time tolerance for a case when the packet was lost and had to be retransmitted. After that time the retransmission is given up, and if the receiver peer still requests this retransmission, it will be denied by DROPREQ message. I don't think it should be described here in such details, but it should be clearly stated that TTL does not concern regular first-time sending of the packet - which is always done regardless of any conditions - but only if any of the packets carrying this message were lost.